### PR TITLE
Add changable Audio Device, bugfixes

### PR DESCRIPTION
**Pull request by heyngra**
Currently implemented:
- [x] Get rid of Bar going out of bounds
- [x] Add settings tab
- [x] Add changable Audio Device

### CODE REVIEW

**Review by CaptSiro — COMMENTED** [review]
I dont think creating a new variable clampFill is necessary. Can you please change it to just using clamp function?

File: src/renderer/src/lib/Music.ts [hunk 1]
```diff
@@ -139,6 +148,11 @@ export function pause() {
   player.pause();
 }
 
+export async function changeAudioDevice(deviceId: string) {
+  await window.api.request("settings::write", "audioDeviceId", deviceId);
+  (player as any).setSinkId(deviceId); // unsafe, but works
```
Comment (CaptSiro) [inline src/renderer/src/lib/Music.ts:153]: Can you change it to proper check? (This code snippet is not tested)
```javascript
if ("setSinkId" in player && typeof player.setSinkId === "function") {
    // ...
}
```

**Review by CaptSiro — COMMENTED** [review]
I would like to see proper checking. Better be sure than sorry. Later I will try to fix the settings part

File: src/renderer/src/lib/Music.ts (hunk 1 above)
Comment (heyngra) [reply src/renderer/src/lib/Music.ts:153]: you can do THAT without everything falling apart? really cool!

**Review by CaptSiro — APPROVED** [review]

File: src/renderer/src/components/settings/SettingsView.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  const setAudioDevicesMap = () => {
+    const audioMap = new Map<string, ()=>any>();
+    navigator.mediaDevices.enumerateDevices().then(r => {
+      for (const device of r) {
+        if (device.kind === "audiooutput") {
+          audioMap.set(device.label, () => changeAudioDevice(device.deviceId));
+        }
+      }
+      setAudioDevices(audioMap);
+      console.log(audioDevices());
```
Comment (CaptSiro) [inline src/renderer/src/components/settings/SettingsView.tsx:21]: Shouldn't this be removed?

Comment (heyngra) [reply src/renderer/src/components/settings/SettingsView.tsx:21]: the console log? yeah, I was supposed to remove it

File: src/renderer/src/components/settings/SettingDropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+import "../../assets/css/select.css";
+
+type SettingDropdownProps = {
+  label: string,
+  options: Map<string, () => any>, // name of option, function after clicked
+  disabled: false,
+}
+
+const settingDropdown: Component<SettingDropdownProps> = props => {
```
Comment (duduBTW) [inline src/renderer/src/components/settings/SettingDropdown.tsx:11]: shouldn't it be `SettingDropdown`?

**Review by CaptSiro — DISMISSED** [review]

File: src/renderer/src/components/settings/SettingDropdown.tsx (hunk 3 above)
Comment (heyngra) [reply src/renderer/src/components/settings/SettingDropdown.tsx]: checked with other files, I think so

**Review by CaptSiro — APPROVED** [review]

**Comment by CaptSiro** [comment]
Thanks for contributing :btmcLuv:

**Pull request merged**